Sources/Tracing/TracingTime.swift

```diff
@@ -0,0 +1,186 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Distributed Tracing open source project
+//
+// Copyright (c) 2020-2023 Apple Inc. and the Swift Distributed Tracing project
+// authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin
+#endif
+
+@_exported import Instrumentation
+@_exported import InstrumentationBaggage
+
+public protocol SwiftDistributedTracingDurationProtocol: Comparable, AdditiveArithmetic, Sendable {
+    static func / (_ lhs: Self, _ rhs: Int) -> Self
+    static func /= (_ lhs: inout Self, _ rhs: Int)
+    static func * (_ lhs: Self, _ rhs: Int) -> Self
+    static func *= (_ lhs: inout Self, _ rhs: Int)
+
+    static func / (_ lhs: Self, _ rhs: Self) -> Double
+}
+
+extension SwiftDistributedTracingDurationProtocol {
+    public static func /= (_ lhs: inout Self, _ rhs: Int) {
+        lhs = lhs / rhs
+    }
+}
+
+public protocol SwiftDistributedTracingInstantProtocol: Comparable, Hashable, Sendable {
+    // associatedtype Duration: SwiftDistributedTracingDurationProtocol
+}
+
+public protocol TracerInstantProtocol: SwiftDistributedTracingInstantProtocol {
+    /// Representation of this instant as the number of milliseconds since UNIX Epoch (January 1st 1970)
+    @inlinable
+    var millisSinceEpoch: UInt64 { get }
```

Comment:
Do we want to spell this out?

```suggestion
    var millisecondsSinceEpoch: UInt64 { get }
```

Reply:
Ok :)